File: source/Knyaz.Optimus.Tests/Resources/JsTests/DocumentFragmentTests.js
Test("DocumentFragmentTests", {
    "Clone": {
        run: function () {
            var doc = document.createDocumentFragment();
            var e = document.createElement("div");
            doc.appendChild(e);
            var deepClone = doc.cloneNode(true);
            Assert.AreEqual("<div></div>", deepClone.firstChild.outerHTML.toLowerCase());
            
            var clone = doc.cloneNode();
            Assert.IsNotNull(clone);
            //Assert.AreEqual(0, clone.childNodes.length);
        }
    }
});

File: source/Knyaz.Optimus.UnitTests/Dom/DocumentFragmentTests.cs
using Knyaz.Optimus.Dom;
using Knyaz.Optimus.Dom.Elements;
using NUnit.Framework;

namespace Knyaz.Optimus.Tests.Dom
{
	[TestFixture]
	class DocumentFragmentTests
	{
		[Test]
		public void CreateDocumentFragment()
		{
			var document = new Document();
			var x = document.CreateDocumentFragment();
			var el = document.CreateElement("div");
			x.AppendChild(el);
			Assert.AreEqual(document, el.OwnerDocument);
		}

		[Test]
		public void CloneDocumentFragment()
		{
			var document = new Document();
			var docFrag = document.CreateDocumentFragment();
			var el = document.CreateElement("div");
			docFrag.AppendChild(el);
			var clone = docFrag.CloneNode(true);

			Assert.AreEqual(1, clone.ChildNodes.Count);
			Assert.AreEqual("<DIV></DIV>", ((Element)clone.FirstChild).OuterHTML);
		}
	}
}